mupdf appends their desired optimization flags to $CFLAGS this
causes the optimization level configured by abuild to be overwritten.

Since abuild uses -Os by default this decreases the package size.

diff -upr mupdf-1.23.3.orig/Makerules mupdf-1.23.3/Makerules
--- mupdf-1.23.3.orig/Makerules	2023-09-09 11:11:42.325742173 +0200
+++ mupdf-1.23.3/Makerules	2023-09-09 11:13:24.952695140 +0200
@@ -92,7 +92,7 @@ ifeq ($(build),debug)
   CFLAGS += -pipe -g
   LDFLAGS += -g
 else ifeq ($(build),release)
-  CFLAGS += -pipe -O2 -DNDEBUG
+  CFLAGS += -pipe -DNDEBUG
   LDFLAGS += $(LDREMOVEUNREACH) -Wl,-s
 else ifeq ($(build),small)
   CFLAGS += -pipe -Os -DNDEBUG
